File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        temp = dict()
        max = 0
        max_value = 0
        
        for key, value in enumerate(nums):
            if value in temp:
                temp[value] = temp[value] + 1 
            else: 
                temp[value] = 1
            
            if temp[value] > max:
                max = temp[value]
                max_value = value
        
        return max_value
        

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        HashMap<Integer, Integer> map = new HashMap<>();
        
        int max = 1;
        Integer max_value = null;
        
        for(int num : nums){
            Integer temp = map.get(num);
            int curCnt = temp == null ? 1 : temp + 1;
            map.put(num, curCnt);
            
            if(curCnt >= max){
                max = curCnt;
                max_value = num;
                
                if(max > nums.length/2){
                    return max_value;
                }
            }
        }
        
        return max_value;
    }
}